examples/reshape/functions/uniqBy.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  AutocompleteReshapeSource,
+  BaseItem,
+} from '@algolia/autocomplete-core';
+
+import { AutocompleteReshapeFunction } from './AutocompleteReshapeFunction';
+import { normalizeReshapeSources } from './normalizeReshapeSources';
+
+type UniqByPredicate<TItem extends BaseItem> = (params: {
+  source: AutocompleteReshapeSource<TItem>;
+  item: TItem;
+}) => TItem;
+
+export const uniqBy: AutocompleteReshapeFunction<UniqByPredicate<any>> = <
+  TItem extends BaseItem
+>(
+  predicate
+) => {
+  return function runUniqBy(...rawSources) {
+    const sources = normalizeReshapeSources(rawSources);
+    const seen: TItem[] = [];
+
+    return sources.map((source) => {
+      const items = source.getItems().filter((item) => {
+        const appliedItem = predicate({ source, item });
+        const hasSeen = seen.includes(appliedItem);
```

Comment:
just an example, but I wonder if a set.has is quicker or slower than array.includes as it should be indexed

Reply:
we're not talking about that many items, so not really important

Reply:
Updated to using sets in c36da9c733803d4c0736e8e6baf6ac7002fbb64b.